File: 4기 코딩테스트_220615/4기 - 2nd 코딩테스트/MostClosestIntegerToZero/MostClosestIntegerToZero_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.medium.MostClosestIntegerToZero.MostClosestIntegerToZero import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        arr = [-5, 2, -1, 3]

        # when
        result = solution(arr)

        # then
        self.assertEqual(-1, result)

    def test_2(self):
        # given
        arr = [-7, 1, 7, -4, 6]

        # when
        result = solution(arr)

        # then
        self.assertEqual(1, result)

    def test_3(self):
        # given
        arr = [-4, 2, -2]

        # when
        result = solution(arr)

        # then
        self.assertEqual(-2, result)

    def test_4(self):
        # given
        arr = [0]

        # when
        result = solution(arr)

        # then
        self.assertEqual(0, result)

    def test_5(self):
        # given
        arr = [100]

        # when
        result = solution(arr)

        # then
        self.assertEqual(100, result)

    def test_6(self):
        # given
        arr = [-100]

        # when
        result = solution(arr)

        # then
        self.assertEqual(-100, result)


if __name__ == '__main__':
    unittest.main()

File: 4기 코딩테스트_220615/4기 - 2nd 코딩테스트/MostClosestIntegerToZero/MostClosestIntegerToZero_test.js
import assert from 'assert'
import solution from './MostClosestIntegerToZero.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const arr = [-5, 2, -1, 3]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(-1, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const arr = [-7, 1, 7, -4, 6]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(1, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const arr = [-4, 2, -2]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(-2, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const arr = [0]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const arr = [100]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(100, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const arr = [-100]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(-100, result)
    done();
  });
})
